sdk/testing/src/main/java/io/opentelemetry/sdk/testing/context/StrictContextStorage.java

```diff
@@ -132,35 +137,34 @@ public Context current() {
    */
   // AssertionError to ensure test runners render the stack trace
   public void ensureAllClosed() {
-    List<CallerStackTrace> leakedCallers = new ArrayList<>();
-    currentCallers.drainTo(leakedCallers);
-    for (CallerStackTrace caller : leakedCallers) {
-      // Sometimes unit test runners truncate the cause of the exception.
-      // This flattens the exception as the caller of close() isn't important vs the one that leaked
-      AssertionError toThrow =
-          new AssertionError(
-              "Thread [" + caller.threadName + "] opened a scope of " + caller.context + " here:");
-      toThrow.setStackTrace(caller.getStackTrace());
-      throw toThrow;
+    pendingScopes.expungeStaleEntries();
+    List<CallerStackTrace> leaked = pendingScopes.drainPendingCallers();
+    if (!leaked.isEmpty()) {
+      if (leaked.size() > 1) {
+        logger.log(Level.SEVERE, "Multiple scopes leaked - first will be thrown as an error.");
+      }
+      for (CallerStackTrace caller : leaked) {
+        logger.log(Level.SEVERE, "Scope leaked", callerError(caller));
```

Comment:
we could skip doing this if there's only one, so you don't get the same one twice? Not sure if that's good or bad, but just a thought.

Reply:
Oops - yeah this is a bug I meant to have it in the if statement, thanks